src/service/action-impl.js

```diff
@@ -566,19 +583,36 @@ export class ActionService {
    * @param {!Array<{tagOrTarget: string, method: string}>} whitelist
    */
   setWhitelist(whitelist) {
-    this.whitelist_ = whitelist;
+    this.whitelist_ = whitelist.filter(
+      (value) => hasOwn(value, 'tagOrTarget') && hasOwn(value, 'method')
+    );
   }
 
   /**
    * Adds an action to the whitelist.
    * @param {string} tagOrTarget The tag or target to whitelist, e.g.
    *     'AMP-LIST', '*'.
    * @param {string} method The method to whitelist, e.g. 'show', 'hide'.
+   * @param {{isAllowedInEmail: boolean}=} opts
    */
-  addToWhitelist(tagOrTarget, method) {
+  addToWhitelist(tagOrTarget, method, opts) {
+    if (
+      opts &&
+      hasOwn(opts, 'isAllowedInEmail') &&
```

Comment:
Note that you don't need to use `hasOwn` instead of dot syntax most of the time. Only if you're worried about prototype collision.